azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DeploymentTemplateBuilderTest.java

```diff
@@ -132,6 +132,41 @@ void testVirtualMachine() {
    }
 
    @Test
+   void testAddStorageResourceWhenNameIsLongerThan24Chars() {
+      String name = "thishasmorethan24characters";
+      DeploymentTemplateBuilder builder = getMockDeploymentTemplateBuilderWithEmptyOptions(name);
+
+      DeploymentBody deploymentBody = builder.getDeploymentTemplate();
+      assertTrue(Iterables.contains(deploymentBody.template().variables().keySet(), "storageAccountName"));
+      String storageAccountName = deploymentBody.template().variables().get("storageAccountName");
+      assertEquals(storageAccountName.length(), 24);
+      assertEquals(storageAccountName.substring(0, 10), name.replaceAll("[^a-z0-9]", "").substring(0, 10));
+      assertEquals(storageAccountName.substring(storageAccountName.length() - 10, storageAccountName.length()), name.replaceAll("[^a-z0-9]", "").substring(name.length() - 10, name.length()));
```

Comment:
Tests should not replicate the business logic (or they aren't actually testing that it produces the expected value; if the business logic is broken, the test is broken too).
Instead, make assertions against concrete values. Assert that the value is a concrete one, or starts with the expected String and has N prefix and N suffix characters, etc.